### ISSUE TITLE
Null coercion not working with property-based `@JsonCreator`

### ISSUE THREAD

**Issue by cowtowncoder** [FasterXML/jackson-databind#2024]
(note: follow-up to #1402 for null coercion)

Although null coercion (and skipping, error reporting) works with fields, setters, it appears not to work as intended via Creator methods. Would be good if it did.
This occurs up to `2.9.5`.

**Comment by cowtowncoder** [comment]
Looks like problem is that `PropertyMetadata` is created differently for `CreatorProperty`; if setter/nulls info was available, empty/fail would work (skip has semantics that can not be supported with creator).
Fundamentally problem seems to be that construction of creator properties proceeds quite differently, and unfortunately full logic that `POJOPropertyBuilder` has, nicely encapsulated, can not be accessed for creator properties (unless there are matching fields, getters, in which case this probably works).

So. The place where this can be done is either in `constructCreatorProperty()` of `BasicDeserializerFactory` (explicitly introspect), or, possibly, in `addBeanProps()` of `BeanDeserializerFactory` (merge, as it seems that metadata does somehow exist?)

**Comment by sanyarnd** [comment]
I have the exact same issue using `lombok`'s `@Value` annotation: `.setDefaultSetterInfo(...);` call is ignored, I hadn't notice any changes in program's behavior with different values of `Nulls.<...>` (error, default).

The workaround I've applied ([source](https://github.com/FasterXML/jackson-databind/issues/347#issuecomment-213153541)):
```java
private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new SimpleModule().setDeserializerModifier(new NullCollectionsAsEmptyModifier()));

private static abstract class ContextualJsonDeserializer extends JsonDeserializer<Object> implements ContextualDeserializer { }
private static class NullCollectionsAsEmptyModifier extends BeanDeserializerModifier {
	@Override
	public JsonDeserializer<?> modifyMapDeserializer(DeserializationConfig config, MapType type, BeanDescription beanDesc, JsonDeserializer<?> deserializer) {
		return new ContextualJsonDeserializer() {
			@Override
			public JsonDeserializer<?> createContextual(DeserializationContext ctxt, BeanProperty property) throws JsonMappingException {
				return modifyMapDeserializer(config, type, beanDesc, ((ContextualDeserializer) deserializer).createContextual(ctxt, property));
			}

			@Override
			public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
				return deserializer.deserialize(p, ctxt);
			}

			@Override
			public Object getNullValue(DeserializationContext ctxt) {
				return Collections.emptyMap();
			}
		};
	}

	@Override
	public JsonDeserializer<?> modifyCollectionDeserializer(DeserializationConfig config, CollectionType type, BeanDescription beanDesc, JsonDeserializer<?> deserializer) {
		return new ContextualJsonDeserializer() {
			@Override
			public JsonDeserializer<?> createContextual(DeserializationContext ctxt, BeanProperty property) throws JsonMappingException {
				return modifyCollectionDeserializer(config, type, beanDesc, ((ContextualDeserializer) deserializer).createContextual(ctxt, property));
			}

			@Override
			public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
				return deserializer.deserialize(p, ctxt);
			}

			@Override
			public Object getNullValue(DeserializationContext ctxt) {
				return Collections.emptyList();
			}
		};
	}
}
```

**Comment by cowtowncoder** [comment]
I just fixed #2458 and I think these two were actually the same issue -- so although this is older, I think I'll close this as duplicate due to release notes adding fixing of the other issue.
Both fixed for 2.10.0.
